phalcon/di.zep

```diff
@@ -205,32 +205,54 @@ class Di implements DiInterface
 	 */
 	public function get(string! name, parameters = null) -> var
 	{
-		var service, eventsManager, instance = null;
+		var service, eventsManager, isShared, instance = null;
+
+		// If the service is shared and it already has a cached instance then
+		// immediately return it without triggering events.
+		if fetch service, this->_services[name] {
+			let isShared = service->isShared();
+			if isShared && isset this->_sharedInstances[name] {
+
+				// FIXME: This must be used due to wasFreshInstance()
+				// @see https://github.com/phalcon/cphalcon/pull/13112
+				return this->getShared(name, parameters);
+
+				// This is the desired simple way of returning the shared instance.
+				// return this->_sharedInstances[name];
+			}
+		}
 
 		let eventsManager = <ManagerInterface> this->_eventsManager;
 
-		if typeof eventsManager == "object" {
+		// Allows for custom creation of instances through the "di:beforeServiceResolve" event.
+		if eventsManager !== null {
```

Comment:
Is it possible for `eventsManager` to have any value other than `<ManagerInterface>` or null? Would it not throw an exception if you tried to set it with anything other than `<ManagerInterface>`?

Reply:
@dschissler Firstly, I do not understand why these changes are needed, because under the hood, completely different mechanisms will be applied. Finally, anyone can change this through inheritance or reflection.

Reply:
@sergeyklay Yeah I wasn't thinking about inheritance and its not a part of an interface.  I will just revert it.  I never felt that strongly about it.  I'll push the revert within the next several hours.